jwst/transforms/schemas/tpcorr-0.7.0.yaml

```diff
@@ -0,0 +1,46 @@
+%YAML 1.1
+---
+$schema: "http://stsci.edu/schemas/yaml-schema/draft-01"
+id: "http://stsci.edu/schemas/jwst_pipeline/tpcorr-0.7.0"
+tag: "tag:stsci.edu:jwst_pipeline/tpcorr-0.7.0"
+title: >
+  Tangent Plane Correction
+
+description: |
+  TPCorr model performs correction of "mis-aligned" (V2, V3) coordinates by:
+  - projecting points with (V2, V3) coordinates onto a tangent plane;
+  - applying linear transformations to the projected coordinates;
+  - projecting points from the tangent plane onto the V2-V3 sphere.
+  Ouput of this model are corrected (V2, V3) coordinates.
+
+anyOf:
+  - $ref: ../asdf/transform/transform-1.1.0
+  - type: object
+    properties:
+      v2ref:
+        description: |
+          V2 reference angle.
+        type: number
+      v3ref:
+        description: |
+          V3 reference angle.
+        type: number
+      roll:
+        description: |
+          Roll angle.
+        type: number
+      matrix:
+        description: |
+          Generalized rotation matrix.
+        type: array
+        items:
+          type: array
+          items:
+            type: number
```

Comment:
This limits the schema to a list of lists. If you know that you are never going to pass a numpy array then it's fine. Or yo ucan pass a ndarray but make sure that you convert it to a list of lists. 

Alternatively you can us `anyOf` and add ndarray as an alternative.

In any case add a test to tags/tests which writes the model to a file and then reads it back in. There are examples in the test file.

Reply:
@nden Thank you! I could not find any examples of how to do this but I changed the code `TPCorrType` to perform the conversion after/before serialization.